infra/main.tf

```diff
@@ -290,12 +293,26 @@ resource "google_compute_global_forwarding_rule" "http" {
   labels                = var.labels
 }
 
-resource "time_sleep" "load_balancer_warm_up_time" {
+# It may take more than 2 minutes for the newly provisioned load balancer
+# to forward requests to the Cloud Run service.  The following data source
+# allows for terraform apply to finish running when the end-point resolves
+
+data "http" "load_balancer_warm_up" {
+  url = "http://${google_compute_global_address.default.address}/"
+  # Attempt retry every 20 seconds 17 times, totaling to a 6 minute timeout.
+  retry {
+    attempts     = 17
+    max_delay_ms = 20000
+    min_delay_ms = 20000
+  }
+  # Begin trying after load balancer resources are created.
   depends_on = [
+    google_compute_global_address.default,
+    google_compute_url_map.default,
+    google_compute_backend_service.default,
```

Comment:
note: `google_compute_url_map.default` should have an implicit dependency on `google_compute_backend_service.default` if you want one fewer explicit dependency here, but clarifying we're waiting on all the load balancer resources may be a good thing given how complicated this is.

Alright, I've talked myself out of this being a suggestion and made it a "note".